enhancements/multi-arch/dynamic-imagestream-importmode-setting.md

```diff
@@ -0,0 +1,408 @@
+
+---
+title: dynamic-imagestream-importmode-setting
+authors:
+  - "@Prashanth684"
+reviewers:
+  - "@deads2k, apiserver"
+  - "@JoelSpeed, API"
+  - "@wking, CVO"
+  - "@dmage, Imagestreams"
+  - "@soltysh, Workloads, oc"
+approvers:
+  - "@deads2k"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2024-04-02
+last-updated: 2024-04-02
+tracking-link:
+  - https://issues.redhat.com/browse/MULTIARCH-4552
+---
+
+# Dynamically set Imagestream importMode based on cluster payload type
+
+## Summary
+
+This is a proposal to set ImportMode for
+imagestreams dynamically based on the payload type
+of the cluster. This means that if the OCP release
+payload is a singlemanifest payload, the
+importMode for all imagestreams by default will be
+`Legacy` and if the payload is a multi payload the
+default will be `PreserveOriginal`. 
+
+## Motivation
+
+This change ensures that imagestreams will be
+compatible for single and multi arch compute
+clusters without needing manual intervention.
+Users who install or upgrade to the multi
+release payload
+(https://mirror.openshift.com/pub/openshift-v4/multi/)
+and want to add compute nodes of differing
+architectures, need not change the behaviour
+of any newly imported imagestreams or any scripts
+which import them. For a cluster installed with
+single arch payload, imagestreams default import
+mode would always be set to `Legacy` because it
+is sufficient if the single image manifest is
+imported and for a cluster installed with the
+multi payload it would be `PreserveOriginal`
+which would import the manifestlist in anticipation
+that users would add/have already added nodes
+of other architectures (which do not match the
+control plane's).
+
+### User Stories
+
+- As an Openshift user, I do not want to manually
+  change the importMode of imagestreams I want to
+  import on the cluster if I add/remove nodes of
+  a different architecture to it.
+- As an Openshift user, I do not want to change my
+  existing scripts and oc commands to add extra
+  flags to enable manifestlist imports on
+  imagestreams.
```

Comment:
yes, this was brought up and I believe this should not be as big of a concern because not a whole lot of users outside of CI and internal tooling consume the image APIs directly